doc/applications.md

```diff
@@ -0,0 +1,151 @@
+# Creating applications
```

Comment:
This file :+1:
But docs are meant to be consumed through reactivex.io/rxjs, not through the Github repo. This file will appear nowhere in the documentation. We need to configure ESDoc first. I recommend putting this file (and also `basics.md`) under what ESDoc calls "Tutorial". Check their instructions for configuring ESDoc tutorial: https://esdoc.org/config.html